### PR TITLE
Add support for custom payment plans. ACDM-1687 #resolve

### PR DESCRIPTION
**Pull request by Luis-Cruz**

### CODE REVIEW

File: src/main/java/org/fenixedu/academic/domain/accounting/Event.java [hunk 1]
```diff
@@ -653,7 +642,7 @@ public Money getTotalAmountToPay() {
     }
 
     public Money getOriginalAmountToPay() {
-        return new Money(getDebtInterestCalculator(getWhenOccured().plusSeconds(1)).getDebtAmount());
+        return new Money(getDebtInterestCalculator(new DateTime()).getDebtAmount());
```
Comment (sergiofbsilva) [inline src/main/java/org/fenixedu/academic/domain/accounting/Event.java]: 😮😮😮 beware 😮😮😮

File: src/main/java/org/fenixedu/academic/domain/accounting/Event.java [hunk 2]
```diff
@@ -18,26 +18,7 @@
...
-import java.util.Optional;
-import java.util.Set;
-import java.util.SortedSet;
-import java.util.TreeSet;
-import java.util.function.BiFunction;
-import java.util.function.Predicate;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
-
+import org.apache.solr.common.util.Hash;
```
Comment (sergiofbsilva) [inline src/main/java/org/fenixedu/academic/domain/accounting/Event.java]: why are we using this import? 🤔

File: src/main/java/org/fenixedu/academic/domain/accounting/Event.java (hunk 1 above)
Comment (Luis-Cruz) [reply src/main/java/org/fenixedu/academic/domain/accounting/Event.java]: The getDebtAmount() originally only iterates the "DebtStream" and all "Debt" objects are always added to the calculator, irregardless of the instance in time being considered; in other words, as long as the sum of values of the inner DueDateAmountMap remains the same then the getDebtAmount() will always produce the same value for every instance in time.

Either way, we ran a script for every existing event to check, and of course the value always matched.

File: src/main/java/org/fenixedu/academic/domain/accounting/Event.java (hunk 2 above)
Comment (Luis-Cruz) [reply src/main/java/org/fenixedu/academic/domain/accounting/Event.java]: It was a mistake.

**Pull request merged**